packages/web-api-scan-runner/src/tasks/notification-queue-message-sender.ts

```diff
@@ -0,0 +1,108 @@
+        @inject(loggerTypes.Process) private readonly currentProcess: typeof process,
+        private readonly system: typeof System = System,
+    ) {}
+
+    public async sendNotificationMessage(onDemandNotificationRequestMessage: OnDemandNotificationRequestMessage): Promise<void> {
+        this.logger.logInfo(`Reading page scan run result ${onDemandNotificationRequestMessage.scanId}`);
+        this.logger.setCustomProperties({
+            scanId: onDemandNotificationRequestMessage.scanId,
+            batchJobId: this.currentProcess.env.AZ_BATCH_JOB_ID,
+        });
+
+        const pageScanResult = await this.enqueueNotificationWithRetry(onDemandNotificationRequestMessage);
+
+        this.logger.logInfo(`Writing page notification status to a storage.`);
+        await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+    }
+
+    private async enqueueNotificationWithRetry(
+        notificationSenderConfigData: OnDemandNotificationRequestMessage,
+    ): Promise<Partial<OnDemandPageScanResult>> {
+        let numberOfTries = 1;
+        let notificationState: NotificationState = 'queueFailed';
+        let error: NotificationError = null;
+        let statusCode = 500;
+        const scanConfig = await this.getScanConfig();
+
+        this.logger.logInfo(`Retry count: ${scanConfig.maxSendNotificationRetryCount}`);
+        while (numberOfTries <= scanConfig.maxSendNotificationRetryCount) {
+            this.logger.logInfo(
+                `Enqueuing the scan notification, try #${numberOfTries} -> retry count ${scanConfig.maxSendNotificationRetryCount}`,
+            );
+            let response;
+            try {
+                response = await this.queue.createMessage(this.storageConfig.notificationQueue, notificationSenderConfigData);
+
+                if (response) {
+                    this.logger.logInfo(
+                        `Notification enqueued successfully!, try #${numberOfTries} -> retry count ${scanConfig.maxSendNotificationRetryCount}`,
+                    );
+                    notificationState = 'queued';
+                    error = null;
+                    statusCode = 200;
+                    break;
+                } else {
+                    this.logger.logInfo(
+                        `Failed to enqueue the notification!, try #${numberOfTries} -> retry count ${scanConfig.maxSendNotificationRetryCount}`,
+                    );
+                    error = { errorType: 'InternalError', message: `Failed to enqueue the notification!` };
+                }
+            } catch (e) {
+                this.logger.logError(`Failed to enqueue the notification!, error message: ${(e as Error).message}`);
+                error = { errorType: 'InternalError', message: (e as Error).message };
+            }
+            numberOfTries = numberOfTries + 1;
+            if (numberOfTries <= scanConfig.maxSendNotificationRetryCount) {
+                await this.system.wait(5000);
```

Comment:
Can we increase max retry count to 5 and remove `maxSendNotificationRetryCount` from all configs b/c it is default value anyway
and
change wait time to the formula: `1000 * (numberOfTries - 1)`